src/celeritas/ext/VecgeomParams.cc

```diff
@@ -144,6 +146,15 @@ VolumeId VecgeomParams::find_volume(Label const& label) const
     return vol_labels_.find(label);
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Locate the volume ID corresponding to a Geant4 logical volume.
+ */
+VolumeId VecgeomParams::find_volume(G4LogicalVolume const* volume) const
+{
+    return g4log_volid_map_.find(volume)->second;
```

Comment:
Add logic: if the volume isn't found (e.g. the map is empty) or the input is null, return `VolumeId{}`.

Reply:
Logic added as suggested. I suggest an ORANGE implementation as well.